snapcraft/_store.py

```diff
@@ -64,6 +64,27 @@ def _get_data_from_snap_file(snap_path):
     return snap_yaml
 
 
+@contextmanager
+def _get_icon_from_snap_file(snap_path):
+    icon_file = None
+    with tempfile.TemporaryDirectory() as temp_dir:
+        output = subprocess.check_output(
+            ['unsquashfs', '-d',
+             os.path.join(temp_dir, 'squashfs-root'),
+             snap_path, '-e', 'meta/gui'])
+        logger.debug("Output extracting icon from snap: %s", output)
+        for extension in ('png', 'svg'):
+            icon_name = 'icon.{}'.format(extension)
+            icon_path = os.path.join(
+                temp_dir, 'squashfs-root', 'meta/gui', icon_name)
+            if os.path.exists(icon_path):
+                icon_file = open(icon_path, 'rb')
+                break
+        yield icon_file
+        if icon_file is not None:
```

Comment:
should the yield above be wrapped in between a `try` and `finally` to always check and close the icon handle?

Reply:
Yes, updating.